src/main/groovy/com/netflix/nebula/lint/rule/dependency/MethodScanner.groovy

```diff
@@ -0,0 +1,101 @@
+package com.netflix.nebula.lint.rule.dependency
+
+import org.objectweb.asm.ClassReader
+import org.objectweb.asm.ClassVisitor
+import org.objectweb.asm.Label
+import org.objectweb.asm.MethodVisitor
+import org.objectweb.asm.Opcodes
+
+import java.nio.file.Path
+
+/**
+ * Scans a given class and retrieves information about method references.
+ * By default, ignores java/lang package.
+ */
+class MethodScanner {
+    private AppClassVisitor classVisitor
+
+    private ArrayList<MethodReference> methodReferences = []
+
+    private class AppMethodVisitor extends MethodVisitor {
+
+        int line
+
+        private final Collection<String> ignoredPackages
+
+        AppMethodVisitor(Collection<String> ignoredPackages) {
+            super(Opcodes.ASM6)
+            this.ignoredPackages = ignoredPackages
+        }
+
+        @Override
+        void visitMethodInsn(int opcode, String owner, String name, String desc, boolean isInterface) {
```

Comment:
What does the `Insn` mean here?

Reply:
it stands for Instruction 😅  https://static.javadoc.io/org.ow2.asm/asm/7.1/org/objectweb/asm/MethodVisitor.html#visitMethodInsn-int-java.lang.String-java.lang.String-java.lang.String-boolean-